### PR TITLE
Bootstrap parity

### LINKED ISSUE #10: Why is there a deploy/contacts.yaml and a kube/kube.yaml?
They've both been modified within roughly the same timeframe. The `make up` target in the Makefile uses `kube/kube.yaml`, but I think the agreed upon location is to put this in`deploy`.

What's going on here? This needs to get cleaned up.

### PR DESCRIPTION
**Pull request by Calebjh**
- Revise project structure to match bootstrap tool generated structure
- Some revision of code to more closely match the bootstrap generated files
- Add a one:many relationship example (e-mails)
- Fix migrations to work with Postgres and augment for emails example support (should be reviewed by somebody with more DB expertise)
- Add a synthetic field, and create example hooks to demonstrate how it can be handled

Resolves #10

### CODE REVIEW

File: pkg/pb/contacts.override.gorm.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package pb
+
+// BeforeToORM ...
```
Comment (dmacthedestroyer) [inline pkg/pb/contacts.override.gorm.go]: if this is hand-written code let's not set a bad example with lazy comments like this :)

File: pkg/pb/contacts.proto [hunk 2]
```diff
@@ -8,15 +8,26 @@ import "github.com/lyft/protoc-gen-validate/validate/validate.proto";
...
+option go_package = "github.com/infobloxopen/atlas-contacts-app/pkg/pb;pb";
 
 message Contact {
     option (gorm.opts).ormable = true;
     uint64 id = 1;
     string first_name = 2;
     string middle_name = 3;
     string last_name = 4;
-    string email_address = 5 [(validate.rules).string.email = true];
+    string primary_email = 6 [(gorm.field).drop = true, (validate.rules).string.email = true];
```
Comment (dmacthedestroyer) [inline pkg/pb/contacts.proto]: great job skipping the 5 -- you're a true protobuffer!

File: pkg/svc/zserver.go [hunk 3]
```diff
@@ -18,9 +18,9 @@ func NewBasicServer(dsn string) (pb.ContactsServer, error) {
 	// In case you didn't have one already.
 	// This is done to make current example up and running
 	// without having contacts DB configured properly.
-	if err := db.AutoMigrate(&pb.ContactORM{}).Error; err != nil {
-		return nil, err
-	}
+	//if err := db.AutoMigrate(&pb.Contact{}).Error; err != nil {
```
Comment (dmacthedestroyer) [inline pkg/svc/zserver.go]: I vote to just remove this code. We don't want to set a precedent of people automigrating their code. We should instead find a different way for them to set up their database

Comment (Calebjh) [reply pkg/svc/zserver.go]: Sounds good. Do we have any process for verifying that the database version and application version are compatible? In lieu of Automigrate, a SchemaCheck might be called for?

Comment (dmacthedestroyer) [reply pkg/svc/zserver.go]: we don't have any process... we should discuss this as a team at some point :)

Comment (dkukharau) [reply pkg/svc/zserver.go]: How will application work without this? We have a DB controller for kubernetes scoped which should be responsible for schema checks and migrations, but that is a big work to do and will not be available in the nearest future.
If remove, please document how to migrate database schema manually, maybe create a `make` target for this.

File: pkg/pb/contacts.proto (hunk 2 above)
Comment (amaskalenka) [reply pkg/pb/contacts.proto]: :+1:

File: pkg/svc/zserver.go (hunk 3 above)
Comment (dmacthedestroyer) [reply pkg/svc/zserver.go]: you're still right, Daniil. We still don't have any way to **initialize** a database. So the process still has some manual steps. My concern with using this `automigrate` as a stopgap is that we'll have multiple projects viewing this as the proper way to do this.

So I think our options are:
1. just have it be painful to initialize a db
2. find some other way to do this easily until a proper solution is developed (the DB controller)

File: Makefile [hunk 4]
```diff
@@ -2,112 +2,108 @@ PROJECT_ROOT		:= github.com/infobloxopen/atlas-contacts-app
...
-GENERATOR            	:= $(DOCKER_RUNNER) $(DOCKER_GENERATOR)
+DOCKER_GENERATOR	:= infoblox/atlas-gentool:v2
+GENERATOR		:= $(DOCKER_RUNNER) $(DOCKER_GENERATOR)
 
-BUILD_TYPE ?= "default"
-ifeq ($(BUILD_TYPE), "default")
-	BUILDER        := $(DOCKER_RUNNER) -w $(SRCROOT_IN_CONTAINER) $(DOCKER_BUILDER)
-endif
+# configuration for the database
+DATBASE_ADDRESS		:= localhost:5432
```
Comment (dkukharau) [inline Makefile]: Typo here?

Comment (dmacthedestroyer) [reply Makefile]: good catch! I think it's supposed to be `DATBASS`. Caleb is all about dat bass... #notreble

**Comment by johnbelamaric** [comment]
groan...

File: Makefile (hunk 4 above)
Comment (Calebjh) [reply Makefile]: Actually, this is propagated from the bootstrap output, raised it with Zach. He confirmed Dan's suspicions about the intended spelling.

**Pull request merged**